devtools/parse_pcap.py

```diff
@@ -0,0 +1,107 @@
+"""Parse pcaps for TP-Link communications."""
+
+import json
+from collections import Counter, defaultdict
+from pprint import pformat as pf
+from pprint import pprint as pp
+
+import click
+import dpkt
+from dpkt.ethernet import ETH_TYPE_IP, Ethernet
+from kasa.protocol import TPLinkSmartHomeProtocol
+
+
+def read_payloads_from_file(file):
+    """Read the given pcap file and yield json payloads."""
+    pcap = dpkt.pcap.Reader(file)
+    for ts, pkt in pcap:
+        eth = Ethernet(pkt)
+        if eth.type != ETH_TYPE_IP:
+            continue
+
+        ip = eth.ip
+        if ip.p == 6:
```

Comment:
Is this attribute access correct? Or the next line?

Reply:
Yeah, dpkt is not very pythonic library, but it's just a dev tool I think it's fine. p = protocol, and 6 & 7 are tcp and udp respectively (just to capture the discoveries too).